### PR TITLE
Fix occasional test failure of `F.average` by changing lower-bound of weight.sum

### PR DESCRIPTION
**Pull request by asi1024**
Fixed the lower bound of `self.w.sum()`.
(#3766)

### CODE REVIEW

File: tests/chainer_tests/functions_tests/math_tests/test_average.py [hunk 1]
```diff
@@ -47,7 +47,8 @@ def setUp(self):
         # Sample weights. Weights should not sum to 0.
         while True:
             self.w = numpy.random.uniform(-2, 2, w_shape).astype(self.dtype)
-            if self.w.sum() > 1e-2:
+            w_sum_eps = 0.2 if self.dtype == numpy.float16 else 5e-2
+            if self.w.sum() > w_sum_eps:
```
Comment (toslunar) [inline tests/chainer_tests/functions_tests/math_tests/test_average.py]: Test `abs(self.w.sum())` to fix the slip in #3704.

Comment (asi1024) [reply tests/chainer_tests/functions_tests/math_tests/test_average.py]: Fixed.

**Comment by kmaehashi** [comment]
I still observe error after applying this PR. Could you repeat-run the unittest test with `@condition.repeat(20000)` annotated to `test_backward_cpu`?

File: tests/chainer_tests/functions_tests/math_tests/test_average.py [hunk 2]
```diff
@@ -47,7 +47,8 @@ def setUp(self):
         # Sample weights. Weights should not sum to 0.
         while True:
             self.w = numpy.random.uniform(-2, 2, w_shape).astype(self.dtype)
-            if self.w.sum() > 1e-2:
+            w_sum_eps = 1.0 if self.dtype == numpy.float16 else 5e-2
+            if abs(self.w.sum()) > w_sum_eps:
```
Comment (kmaehashi) [inline tests/chainer_tests/functions_tests/math_tests/test_average.py]: As discussed offline, could you try sum with high precision?

Comment (asi1024) [reply tests/chainer_tests/functions_tests/math_tests/test_average.py]: I tried `abs(self.w.astype(numpy.float64).sum()) > w_sum_eps`, but the test failed.

Comment (kmaehashi) [reply tests/chainer_tests/functions_tests/math_tests/test_average.py]: I see.

**Comment by kmaehashi** [comment]
LGTM!

**Pull request merged**